### ISSUE TITLE
Workaround for missing jquery for notes and adminbar in AMP mode doesn't work

### ISSUE THREAD

**Issue by gravityrail** [Automattic/jetpack#16775]
#### Steps to reproduce the issue

1. Install the AMP plugin and set it to "standard" mode
2. As a logged-in user, go to any page
3. You will see JS errors like this:

```
notes-common-v2.js?ver=8.8-202032:158 Uncaught ReferenceError: jQuery is not defined
    at notes-common-v2.js?ver=8.8-202032:158
(anonymous) @ notes-common-v2.js?ver=8.8-202032:158
admin-bar-v2.js?ver=8.8-202032:formatted:477 Uncaught ReferenceError: jQuery is not defined
    at admin-bar-v2.js?ver=8.8-202032:formatted:477
```

4. Notifications icon doesn't work

#### What I expected
No JS errors

#### What happened instead
JS errors

#### Possible cause/solution

In `notes.php`, we have [this code](https://github.com/Automattic/jetpack/blob/master/modules/notes.php#L154) which attempts to "allow" script tags in AMP. The problem is, by the time this code runs, the script tags have already been stripped out so it has no effect.

```php
		if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
			add_filter(
				'script_loader_tag',
				function ( $tag, $handle ) use ( $script_handles ) {
					if ( in_array( $handle, $script_handles, true ) ) {
						$tag = preg_replace( '/(?<=<script)(?=\s|>)/i', ' data-ampdevmode', $tag );
					}
					return $tag;
				},
				10,
				2
			);
		}
```

The solution here is to implement a subclass of `AMP_Base_Sanitizer`, or something like that, which explicitly allows those tags if they are truly necessary.

**Comment by westonruter** [comment]
> The problem is, by the time this code runs, the script tags have already been stripped out so it has no effect.

I don't believe this is the case. The `script_loader_tag` filter runs while the template is being rendered. Scripts are only stripped out by the sanitizers after the output buffering has completed for the entire rendered template.

**Comment by westonruter** [comment]
I'm not able to reproduce the issue using the current stable versions of Jetpack and the AMP plugin. I see no JS errors in the console, and I see as expected:

```
<script data-ampdevmode src="https://weston.ruter.net/wp-includes/js/jquery/jquery.js?ver=1.12.4-wp"></script>
```

What version of Core are you noticing this on? What version of Jetpack?
